
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Load data &#8212; Practical tools for quick data visualization</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      
      <h1 class="site-logo" id="site-title">Practical tools for quick data visualization</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="README.html">
   Practical tools for quick data visualization
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Get started
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="01-intro.html">
   Introduction
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Personalize the details
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="02-tidytuesday.html">
   The R data viz community
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Biology-related viz
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="03-bio.html">
   Biology-related data visualization
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  A multi-omics exercise
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="04-exercise.html">
   Exploratory Data Analysis of the most popular human gene
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/intro.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/MiqG/practical_tools_for_quick_data_visualization"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/MiqG/practical_tools_for_quick_data_visualization/issues/new?title=Issue%20on%20page%20%2Fintro.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Load data
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-plotting-with-ggpubr">
   General plotting with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggpubr
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how-many-penguins-of-each-species-did-we-observe-in-total">
     How many penguins of each species did we observe in total?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how-many-penguins-of-each-species-and-sex-did-we-observe-across-the-different-islands">
     How many penguins of each species and sex did we observe across the different islands?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-are-the-distributions-of-flipper-lengths-considering-penguin-species-sex-and-islands-of-origin">
     What are the distributions of flipper lengths considering penguin species, sex and islands of origin?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#are-the-differences-of-body-mass-between-sexes-significant-if-we-control-for-species-and-island">
     Are the differences of body mass between sexes significant if we control for species and island?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-is-the-relationship-between-flipper-length-body-mass-and-bill-length">
     What is the relationship between flipper length, body mass and bill length?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#could-we-have-sampling-bias-in-the-relationship-between-flipper-length-and-body-mass">
     Could we have sampling bias in the relationship between flipper length and body mass?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-is-the-spearman-correlation-coefficient-between-body-mass-and-flipper-length">
     What is the spearman correlation coefficient between body mass and flipper length?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#create-and-save-a-figure">
     Create and save a figure
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#heatmaps-with-pheatmap">
   Heatmaps with
   <code class="docutils literal notranslate">
    <span class="pre">
     pheatmap
    </span>
   </code>
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">
   Phylogenetic trees with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggtree
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#basic-tree">
     Basic tree
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
     Tree and multiple sequence alignment of
     <em>
      TP53
     </em>
     amino acids
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#session-info">
   Session Info
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Load data</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Load data
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#general-plotting-with-ggpubr">
   General plotting with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggpubr
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how-many-penguins-of-each-species-did-we-observe-in-total">
     How many penguins of each species did we observe in total?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how-many-penguins-of-each-species-and-sex-did-we-observe-across-the-different-islands">
     How many penguins of each species and sex did we observe across the different islands?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-are-the-distributions-of-flipper-lengths-considering-penguin-species-sex-and-islands-of-origin">
     What are the distributions of flipper lengths considering penguin species, sex and islands of origin?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#are-the-differences-of-body-mass-between-sexes-significant-if-we-control-for-species-and-island">
     Are the differences of body mass between sexes significant if we control for species and island?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-is-the-relationship-between-flipper-length-body-mass-and-bill-length">
     What is the relationship between flipper length, body mass and bill length?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#could-we-have-sampling-bias-in-the-relationship-between-flipper-length-and-body-mass">
     Could we have sampling bias in the relationship between flipper length and body mass?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-is-the-spearman-correlation-coefficient-between-body-mass-and-flipper-length">
     What is the spearman correlation coefficient between body mass and flipper length?
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#create-and-save-a-figure">
     Create and save a figure
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#heatmaps-with-pheatmap">
   Heatmaps with
   <code class="docutils literal notranslate">
    <span class="pre">
     pheatmap
    </span>
   </code>
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">
   Phylogenetic trees with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggtree
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#basic-tree">
     Basic tree
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
     Tree and multiple sequence alignment of
     <em>
      TP53
     </em>
     amino acids
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#session-info">
   Session Info
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="load-data">
<h1>Load data<a class="headerlink" href="#load-data" title="Permalink to this headline">¶</a></h1>
<a class="reference internal image-reference" href="_images/palmerpenguins.png"><img alt="_images/palmerpenguins.png" src="_images/palmerpenguins.png" style="width: 25%;" /></a>
<p>To start playing around with the functions from these packages we will
use the
<a class="reference external" href="https://allisonhorst.github.io/palmerpenguins/articles/intro.html"><code class="docutils literal notranslate"><span class="pre">palmerpenguins</span></code></a>
data set. This simple data set has both continuous and categorical
variables that make it perfect for showcasing how different functions
work.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(dplyr)
require(tidyr)

penguins_url = &#39;https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/inst/extdata/penguins.csv&#39;
dat = read.csv(url(penguins_url), stringsAsFactors = TRUE)
dat = dat %&gt;% drop_na()
head(dat)

##   species    island bill_length_mm bill_depth_mm flipper_length_mm body_mass_g
## 1  Adelie Torgersen           39.1          18.7               181        3750
## 2  Adelie Torgersen           39.5          17.4               186        3800
## 3  Adelie Torgersen           40.3          18.0               195        3250
## 4  Adelie Torgersen           36.7          19.3               193        3450
## 5  Adelie Torgersen           39.3          20.6               190        3650
## 6  Adelie Torgersen           38.9          17.8               181        3625
##      sex year
## 1   male 2007
## 2 female 2007
## 3 female 2007
## 4 female 2007
## 5   male 2007
## 6 female 2007
</pre></div>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="general-plotting-with-ggpubr">
<h1>General plotting with <code class="docutils literal notranslate"><span class="pre">ggpubr</span></code><a class="headerlink" href="#general-plotting-with-ggpubr" title="Permalink to this headline">¶</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ggpubr)
require(ggplot2)
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">ggpubr</span></code> allows to make insightful plots quickly for exploration that in
turn can be further customized thanks to being built on top of
<code class="docutils literal notranslate"><span class="pre">ggplot2</span></code>.</p>
<p>These are useful links for using this package:</p>
<ul class="simple">
<li><p><a class="reference external" href="http://rpkgs.datanovia.com/ggpubr/reference/index.html">function
reference</a></p></li>
<li><p><a class="reference external" href="http://rpkgs.datanovia.com/ggpubr/reference/get_palette.html">palettes</a></p></li>
<li><p><a class="reference external" href="http://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">figure
making</a></p></li>
</ul>
<p>Next, we will try to answer different questions using this library and
<code class="docutils literal notranslate"><span class="pre">ggplot2</span></code>.</p>
<div class="section" id="how-many-penguins-of-each-species-did-we-observe-in-total">
<h2>How many penguins of each species did we observe in total?<a class="headerlink" href="#how-many-penguins-of-each-species-did-we-observe-in-total" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggpie(dat %&gt;% count(species), x = &#39;n&#39;, fill = &#39;species&#39;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-3-11.png" /></p>
</div>
<div class="section" id="how-many-penguins-of-each-species-and-sex-did-we-observe-across-the-different-islands">
<h2>How many penguins of each species and sex did we observe across the different islands?<a class="headerlink" href="#how-many-penguins-of-each-species-and-sex-did-we-observe-across-the-different-islands" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggbarplot(dat %&gt;% count(species, sex, island), x = &#39;species&#39;, y = &#39;n&#39;, fill = &#39;sex&#39;, 
          label = TRUE, position = position_dodge(0.7), facet.by = &#39;island&#39;, palette = &#39;lancet&#39;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-4-12.png" /></p>
</div>
<div class="section" id="what-are-the-distributions-of-flipper-lengths-considering-penguin-species-sex-and-islands-of-origin">
<h2>What are the distributions of flipper lengths considering penguin species, sex and islands of origin?<a class="headerlink" href="#what-are-the-distributions-of-flipper-lengths-considering-penguin-species-sex-and-islands-of-origin" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>gghistogram(dat, x = &#39;flipper_length_mm&#39;, fill = &#39;sex&#39;, facet.by = c(&#39;species&#39;,&#39;island&#39;))
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-5-12.png" /></p>
<p>Alternatively, we can use stripcharts charts:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggstripchart(dat, x = &#39;island&#39;, y = &#39;flipper_length_mm&#39;, color = &#39;sex&#39;, facet.by = &#39;species&#39;, alpha = 0.5, position = position_jitterdodge(), add = &#39;median_iqr&#39;, add.params = list(color=&#39;black&#39;, group=&#39;sex&#39;, size=0.2))
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-6-12.png" /></p>
</div>
<div class="section" id="are-the-differences-of-body-mass-between-sexes-significant-if-we-control-for-species-and-island">
<h2>Are the differences of body mass between sexes significant if we control for species and island?<a class="headerlink" href="#are-the-differences-of-body-mass-between-sexes-significant-if-we-control-for-species-and-island" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggstripchart(dat, x = &#39;island&#39;, y = &#39;body_mass_g&#39;, color = &#39;sex&#39;, facet.by = &#39;species&#39;, alpha = 0.5, position = position_jitterdodge(), add = &#39;median_iqr&#39;, add.params = list(color=&#39;black&#39;, group=&#39;sex&#39;, size=0.2))+
   stat_compare_means(aes(color = sex), label = &quot;p.signif&quot;, method = &#39;wilcox.test&#39;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-7-12.png" /></p>
</div>
<div class="section" id="what-is-the-relationship-between-flipper-length-body-mass-and-bill-length">
<h2>What is the relationship between flipper length, body mass and bill length?<a class="headerlink" href="#what-is-the-relationship-between-flipper-length-body-mass-and-bill-length" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggscatter(dat, x = &#39;flipper_length_mm&#39;, y = &#39;body_mass_g&#39;, color = &#39;bill_length_mm&#39;, alpha = 0.5)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-8-14.png" /></p>
</div>
<div class="section" id="could-we-have-sampling-bias-in-the-relationship-between-flipper-length-and-body-mass">
<h2>Could we have sampling bias in the relationship between flipper length and body mass?<a class="headerlink" href="#could-we-have-sampling-bias-in-the-relationship-between-flipper-length-and-body-mass" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggscatter(dat %&gt;% mutate(year=factor(year)), x = &#39;flipper_length_mm&#39;, y = &#39;body_mass_g&#39;, alpha = 0.5, color = &#39;year&#39;, ellipse = TRUE)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-9-13.png" /></p>
</div>
<div class="section" id="what-is-the-spearman-correlation-coefficient-between-body-mass-and-flipper-length">
<h2>What is the spearman correlation coefficient between body mass and flipper length?<a class="headerlink" href="#what-is-the-spearman-correlation-coefficient-between-body-mass-and-flipper-length" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>ggscatter(dat %&gt;% mutate(year=factor(year)), x = &#39;flipper_length_mm&#39;, y = &#39;body_mass_g&#39;, alpha = 0.5, color = &#39;year&#39;, 
          add = &#39;reg.line&#39;, conf.int = TRUE, 
          cor.coef = TRUE,
          cor.coeff.args = list(method = &#39;spearman&#39;, label.sep = &#39;\n&#39;)) + 
   theme(aspect.ratio = 1)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-10-11.png" /></p>
</div>
<div class="section" id="create-and-save-a-figure">
<h2>Create and save a figure<a class="headerlink" href="#create-and-save-a-figure" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>fontsize = 6
labsize = 2

# overview number of observations of every sex across islands and species
p1 = ggbarplot(dat %&gt;% count(species, sex, island), x = &#39;species&#39;, y = &#39;n&#39;, fill = &#39;sex&#39;, 
               label = TRUE, lab.size = labsize,
               position = position_dodge(0.7), facet.by = &#39;island&#39;, palette = &#39;lancet&#39;) + 
   ylim(NA, 68)

# sex-related body mass distributions across islands and species
p2 = ggstripchart(dat, x = &#39;island&#39;, y = &#39;body_mass_g&#39;, color = &#39;sex&#39;, facet.by = &#39;species&#39;, 
                  alpha = 0.5, position = position_jitterdodge(), add = &#39;median_iqr&#39;, 
                  add.params = list(color=&#39;black&#39;, group=&#39;sex&#39;, size=0.2),
                  palette = &#39;lancet&#39;)+
            stat_compare_means(aes(color = sex), label = &quot;p.signif&quot;, method = &#39;wilcox.test&#39;, size = labsize)

# association of flipper length and body mass
p3 = ggscatter(dat %&gt;% mutate(year=factor(year)), x = &#39;flipper_length_mm&#39;, y = &#39;body_mass_g&#39;, alpha = 0.5, color = &#39;year&#39;, 
          add = &#39;reg.line&#39;, conf.int = TRUE, 
          cor.coef = TRUE,
          cor.coeff.args = list(method = &#39;spearman&#39;, label.sep = &#39;\n&#39;, size = labsize)) + 
   theme(aspect.ratio = 1)


p1p2 = ggarrange(p1 + theme_pubr(base_size = fontsize), p2 + theme_pubr(base_size = fontsize), ncol = 1, common.legend = TRUE)
fig = ggarrange(p1p2, p3 + theme_pubr(base_size = fontsize), widths = c(2,1), heights = c(2, 1), labels = &#39;AUTO&#39;)

# save
ggsave(&#39;images/myfig.png&#39;, fig, width = 15, height = 10, unit = &#39;cm&#39;)
</pre></div>
</div>
<p><img alt="" src="_images/myfig.png" /></p>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="heatmaps-with-pheatmap">
<h1>Heatmaps with <code class="docutils literal notranslate"><span class="pre">pheatmap</span></code><a class="headerlink" href="#heatmaps-with-pheatmap" title="Permalink to this headline">¶</a></h1>
<p>A part from <code class="docutils literal notranslate"><span class="pre">ggpubr</span></code>, one of the most common packages to visualize
multiple types of data altogether is <code class="docutils literal notranslate"><span class="pre">pheatmap</span></code>, which allows to combine
hierarchical clustering of rows and columns with continuous and
categorical data.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(pheatmap)
# we are only interested in numeric columns
cols_oi = c(&#39;bill_length_mm&#39;,&#39;bill_depth_mm&#39;,&#39;flipper_length_mm&#39;,&#39;body_mass_g&#39;)
rownames(dat) = 1:nrow(dat)
pheatmap(dat[,cols_oi], scale = &#39;column&#39;, show_rownames = FALSE, annotation_row = dat[,c(&#39;species&#39;,&#39;island&#39;),drop=FALSE])
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-12-11.png" /></p>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="phylogenetic-trees-with-ggtree">
<h1>Phylogenetic trees with <code class="docutils literal notranslate"><span class="pre">ggtree</span></code><a class="headerlink" href="#phylogenetic-trees-with-ggtree" title="Permalink to this headline">¶</a></h1>
<p>Trees are another very common plot to visualize hierarchical patterns.
Here, we will use
<a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a>
to visualize the phylogenetic relationships extracted from a multiple
sequence alignment of <em>TP53</em> across mammals.</p>
<div class="section" id="basic-tree">
<h2>Basic tree<a class="headerlink" href="#basic-tree" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ggtree)
set.seed(100)
tree &lt;- rtree(50)
ggtree(tree, layout=&quot;circular&quot;) + geom_tiplab()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-13-12.png" /></p>
</div>
<div class="section" id="tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
<h2>Tree and multiple sequence alignment of <em>TP53</em> amino acids<a class="headerlink" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids" title="Permalink to this headline">¶</a></h2>
<p>We will follow <a class="reference external" href="https://github.com/RussellGrayxd/Phylogenetics/blob/master/Sequence%20analysis.R">Russell J.
Gray</a>’s
approach.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(seqinr)
require(ape)

# create tree from alignment
fasta_file = file.path(here::here(), &#39;data&#39;, &#39;raw&#39;, &#39;TP53-mammals-alignment-aa.fa&#39;)
aln = read.alignment(fasta_file, format = &#39;fasta&#39;, whole.header=TRUE)
D = dist.alignment(aln)
tree = njs(D)

# plot tree with MSA
tree_plot = ggtree(tree)
msaplot(tree_plot, fasta = fasta_file) + ggtitle(&#39;tree with MSA&#39;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-14-1.png" /></p>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="references">
<h1>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="https://ggplot2.tidyverse.org/"><code class="docutils literal notranslate"><span class="pre">ggplot2</span></code></a></p></li>
<li><p><a class="reference external" href="https://rpkgs.datanovia.com/ggpubr/"><code class="docutils literal notranslate"><span class="pre">ggpubr</span></code></a></p></li>
<li><p><a class="reference external" href="https://cran.r-project.org/web/packages/pheatmap/index.html"><code class="docutils literal notranslate"><span class="pre">pheatmap</span></code></a></p></li>
<li><p><a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a></p></li>
<li><p><a class="reference external" href="https://seqinr.r-forge.r-project.org/"><code class="docutils literal notranslate"><span class="pre">seqinr</span></code></a></p></li>
<li><p><a class="reference external" href="https://www.rdocumentation.org/packages/ape/versions/5.4-1"><code class="docutils literal notranslate"><span class="pre">ape</span></code></a></p></li>
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html"><code class="docutils literal notranslate"><span class="pre">Biostrings</span></code></a></p></li>
<li><p><a class="reference external" href="https://github.com/RussellGrayxd"><code class="docutils literal notranslate"><span class="pre">Russell</span> <span class="pre">J.</span> <span class="pre">Gray's</span> <span class="pre">github</span></code></a></p></li>
</ul>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="session-info">
<h1>Session Info<a class="headerlink" href="#session-info" title="Permalink to this headline">¶</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>sessionInfo()

## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.5 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ape_5.4-1       seqinr_4.2-5    ggtree_2.4.1    pheatmap_1.0.12
## [5] ggpubr_0.4.0    ggplot2_3.3.3   tidyr_1.1.2     dplyr_1.0.4    
## 
## loaded via a namespace (and not attached):
##  [1] jsonlite_1.7.2      splines_4.0.3       carData_3.0-4      
##  [4] here_1.0.1          stats4_4.0.3        BiocManager_1.30.10
##  [7] rvcheck_0.1.8       highr_0.8           cellranger_1.1.0   
## [10] yaml_2.2.1          progress_1.2.2      pillar_1.4.7       
## [13] backports_1.2.1     lattice_0.20-41     glue_1.4.2         
## [16] digest_0.6.27       XVector_0.30.0      RColorBrewer_1.1-2 
## [19] ggsignif_0.6.0      colorspace_2.0-0    cowplot_1.1.1      
## [22] htmltools_0.5.1.1   Matrix_1.3-2        pkgconfig_2.0.3    
## [25] broom_0.7.4         haven_2.3.1         zlibbioc_1.36.0    
## [28] purrr_0.3.4         patchwork_1.1.1     tidytree_0.3.3     
## [31] scales_1.1.1        openxlsx_4.2.3      rio_0.5.16         
## [34] tibble_3.0.6        mgcv_1.8-33         IRanges_2.24.1     
## [37] generics_0.1.0      farver_2.0.3        car_3.0-10         
## [40] ellipsis_0.3.1      withr_2.4.1         BiocGenerics_0.36.0
## [43] lazyeval_0.2.2      magrittr_2.0.1      crayon_1.4.1       
## [46] readxl_1.3.1        evaluate_0.14       nlme_3.1-152       
## [49] MASS_7.3-53         rstatix_0.6.0       forcats_0.5.1      
## [52] foreign_0.8-81      tools_4.0.3         data.table_1.13.6  
## [55] prettyunits_1.1.1   hms_1.0.0           lifecycle_0.2.0    
## [58] stringr_1.4.0       S4Vectors_0.28.1    aplot_0.0.6        
## [61] munsell_0.5.0       ggsci_2.9           zip_2.1.1          
## [64] Biostrings_2.58.0   ade4_1.7-16         compiler_4.0.3     
## [67] rlang_0.4.10        grid_4.0.3          labeling_0.4.2     
## [70] rmarkdown_2.6       gtable_0.3.0        abind_1.4-5        
## [73] curl_4.3            R6_2.5.0            gridExtra_2.3      
## [76] knitr_1.31          rprojroot_2.0.2     treeio_1.14.3      
## [79] stringi_1.5.3       parallel_4.0.3      Rcpp_1.0.6         
## [82] vctrs_0.3.6         tidyselect_1.1.0    xfun_0.20
</pre></div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Miquel Anglada Girotto (miquel [dot] anglada [at] crg [dot] eu)<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>