---
title: Introduction
author: miquel [dot] anglada [at] crg [dot] eu
date: 2021-02-17

font-size: 11pt
geometry: margin=1.0in
urlcolor: blue

output: md_document
---
# Load data

<img src='images/palmerpenguins.png' width='25%'>

To start playing around with the functions from these packages we will use the [`palmerpenguins`](https://allisonhorst.github.io/palmerpenguins/articles/intro.html) data set.
This simple data set has both continuous and categorical variables that make it perfect for showcasing how different functions work.


```{r, message=FALSE, warning=FALSE}
require(dplyr)
require(tidyr)

penguins_url = 'https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/inst/extdata/penguins.csv'
dat = read.csv(url(penguins_url), stringsAsFactors = TRUE)
dat = dat %>% drop_na()
head(dat)
```

# General plotting with `ggpubr`
```{r message=FALSE, warning=FALSE}
require(ggpubr)
require(ggplot2)
```

`ggpubr` allows to make insightful plots quickly for exploration that in turn can be further customized thanks to being built on top of `ggplot2`.

These are useful links for using this package:

- [function reference](http://rpkgs.datanovia.com/ggpubr/reference/index.html)
- [palettes](http://rpkgs.datanovia.com/ggpubr/reference/get_palette.html)
- [figure making](http://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html)

Next, we will try to answer different questions using this library and `ggplot2`.

## How many penguins of each species did we observe in total?
```{r}
# try ggpie()
ggpie(dat %>% count(species), x = 'n', fill = 'species')
```
## How many penguins of each species and sex did we observe across the different islands?
```{r}
# try ggbarplot()
ggbarplot(dat %>% count(species, sex, island), x = 'species', y = 'n', fill = 'sex', facet.by='island', 
          label=TRUE, position = position_dodge(0.7), palette = 'simpsons')
```

## What are the distributions of flipper lengths considering penguin species, sex and islands of origin?
```{r}
# try gghistogram()
gghistogram(dat, x = 'flipper_length_mm', fill = 'sex', facet.by = c('species','island'))
```
Alternatively, we can use stripcharts charts:
```{r}
# try ggstripchart()
plt = ggstripchart(dat, x = 'species', y = 'flipper_length_mm', color = 'sex', facet.by = 'island', position = position_jitterdodge(),
             add = 'median_iqr', add.params = list(color='black', group='sex', size=0.2), alpha=0.5)
plt
```
## Are the differences of body mass between sexes significant if we control for species and island?
```{r}
# try ggstripchart() with stat_compare_means()
plt + stat_compare_means(aes(color=sex), label = 'p.signif', method = 'wilcox.test')
```

## What is the relationship between flipper length, body mass and bill length?
```{r}
# try ggscatter()
ggscatter(dat, x = 'bill_length_mm', y = 'flipper_length_mm', color = 'body_mass_g', alpha = 0.5)
```

## Could we have sampling bias in the relationship between flipper length and body mass?
```{r}
# try ggscatter()'s 'color' options
ggscatter(dat %>% mutate(year=as.factor(year)), x = 'bill_length_mm', y = 'flipper_length_mm', color = 'year', alpha = 0.5, ellipse = TRUE)
```

## What is the spearman correlation coefficient between body mass and flipper length?
```{r, message=FALSE, warning=FALSE}
# try ggscatter() with 'cor.coef' and 'add' options
ggscatter(dat %>% mutate(year=as.factor(year)), x = 'bill_length_mm', y = 'flipper_length_mm', color = 'year', alpha = 0.5, add = 'reg.line') + stat_cor(aes(color=year))
```

## Create and save a figure
```{r, warning=FALSE, message=FALSE}
# try ggarrange() and ggsave()
p1 = ggscatter(dat %>% mutate(year=as.factor(year)), x = 'bill_length_mm', y = 'flipper_length_mm', color = 'year', alpha = 0.5, add = 'reg.line') + stat_cor(aes(color=year))
p2 = plt + stat_compare_means(aes(color=sex), label = 'p.signif', method = 'wilcox.test')
p3 = ggpie(dat %>% count(species), x = 'n', fill = 'species')

out = ggarrange(p1,p2, ncol = 1)
out2 = ggarrange(out, p3)

ggsave('images/out.png',out, height = 10, width = 5)
```

![](images/myfig.png)

# Heatmaps with `pheatmap`
A part from `ggpubr`, one of the most common packages to visualize multiple types of data altogether is `pheatmap`, which allows to combine hierarchical clustering of rows and columns with continuous and categorical data.

```{r message=FALSE, warning=FALSE}
require(pheatmap)
rownames(dat) = 1:nrow(dat)
cols_oi = c('bill_length_mm','flipper_length_mm','body_mass_g')
pheatmap(dat[,cols_oi], show_rownames = FALSE, annotation_row = dat[,c('species','island'),drop=FALSE], scale = 'column')
```

# Phylogenetic trees with `ggtree`
Trees are another very common plot to visualize hierarchical patterns. Here, we will use [`ggtree`](https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html) to visualize the phylogenetic relationships extracted from a multiple sequence alignment of *TP53* across mammals.

## Basic tree
```{r, message=FALSE, warning=FALSE}
# create a random rtree with rtree()
# plot with ggtree()
require(ggtree)
tree = rtree(50)
ggtree(tree, layout = 'circular') + geom_tiplab()
```

## Tree and multiple sequence alignment of *TP53* amino acids
We will follow [Russell J. Gray](https://github.com/RussellGrayxd/Phylogenetics/blob/master/Sequence%20analysis.R)'s approach.
```{r, warning=FALSE, message=FALSE, fig.height=10}
# create tree from alignment
fasta_file = file.path(here::here(), 'data', 'raw', 'TP53-mammals-alignment-aa.fa')
aln = read.alignment(fasta_file, format = 'fasta', whole.header=TRUE)
D = dist.alignment(aln)
tree = njs(D)

# plot with msaplot()
```


# References
- [`ggplot2`](https://ggplot2.tidyverse.org/)
- [`ggpubr`](https://rpkgs.datanovia.com/ggpubr/)
- [`pheatmap`](https://cran.r-project.org/web/packages/pheatmap/index.html)
- [`ggtree`](https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html)
- [`seqinr`](https://seqinr.r-forge.r-project.org/)
- [`ape`](https://www.rdocumentation.org/packages/ape/versions/5.4-1)
- [`Biostrings`](https://bioconductor.org/packages/release/bioc/html/Biostrings.html)
- [`Russell J. Gray's github`](https://github.com/RussellGrayxd)

# Session Info
```{r session info}
sessionInfo()
```