
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Biology-related data visualization &#8212; Practical tools for quick data visualization</title>
    
  <link href="_static/css/theme.css" rel="stylesheet">
  <link href="_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Exploratory Data Analysis of the most popular human gene" href="04-exercise.html" />
    <link rel="prev" title="Customize every detail" href="02-tidytuesday.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      
      <h1 class="site-logo" id="site-title">Practical tools for quick data visualization</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="README.html">
   Practical tools for quick data visualization
  </a>
 </li>
</ul>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="01-intro.html">
   Get started
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="02-tidytuesday.html">
   Customize every detail
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Biology-related data visualization
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="04-exercise.html">
   Exploratory Data Analysis of the most popular human gene
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/03-bio.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/MiqG/practical_tools_for_quick_data_visualization"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/MiqG/practical_tools_for_quick_data_visualization/issues/new?title=Issue%20on%20page%20%2F03-bio.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Biology-related data visualization
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler">
   Gene Set Enrichment Analysis (GSEA) with
   <code class="docutils literal notranslate">
    <span class="pre">
     clusterProfiler
    </span>
   </code>
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">
   Phylogenetic trees with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggtree
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#basic-tree">
     Basic tree
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
     Tree and multiple sequence alignment of
     <em>
      TP53
     </em>
     amino acids
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#session-info">
   Session Info
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Biology-related data visualization</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Biology-related data visualization
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler">
   Gene Set Enrichment Analysis (GSEA) with
   <code class="docutils literal notranslate">
    <span class="pre">
     clusterProfiler
    </span>
   </code>
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">
   Phylogenetic trees with
   <code class="docutils literal notranslate">
    <span class="pre">
     ggtree
    </span>
   </code>
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#basic-tree">
     Basic tree
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
     Tree and multiple sequence alignment of
     <em>
      TP53
     </em>
     amino acids
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#session-info">
   Session Info
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="biology-related-data-visualization">
<h1>Biology-related data visualization<a class="headerlink" href="#biology-related-data-visualization" title="Permalink to this headline">¶</a></h1>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="gene-set-enrichment-analysis-gsea-with-clusterprofiler">
<h1>Gene Set Enrichment Analysis (GSEA) with <code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code><a class="headerlink" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler" title="Permalink to this headline">¶</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(tidyverse)
</pre></div>
</div>
<p>Let’s say we have found a set of interesting genes.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>geneset_file = file.path(here::here(), &quot;data&quot;, &quot;raw&quot;, &quot;geneset.txt&quot;)
genes_oi = readLines(geneset_file)

genes_oi %&gt;% length() %&gt;% print()

## [1] 166

genes_oi %&gt;% head()

## [1] &quot;ADAM10&quot; &quot;ADAM17&quot; &quot;APAF1&quot;  &quot;APBB1&quot;  &quot;APH1A&quot;  &quot;APOE&quot;
</pre></div>
</div>
<p>Is is enriched in some biological processes?</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(clusterProfiler)
require(org.Hs.eg.db)

result = enrichGO(genes_oi, ont=&quot;BP&quot;, keyType=&quot;SYMBOL&quot;, OrgDb=org.Hs.eg.db)
result %&gt;% as.data.frame() %&gt;% head()

##                    ID                              Description GeneRatio
## GO:0006119 GO:0006119                oxidative phosphorylation    75/156
## GO:0009060 GO:0009060                      aerobic respiration    77/156
## GO:0046034 GO:0046034                    ATP metabolic process    82/156
## GO:0045333 GO:0045333                     cellular respiration    78/156
## GO:0019646 GO:0019646         aerobic electron transport chain    60/156
## GO:0042773 GO:0042773 ATP synthesis coupled electron transport    61/156
##              BgRatio        pvalue      p.adjust        qvalue
## GO:0006119 141/18722 5.925247e-125 1.667957e-121 1.118312e-121
## GO:0009060 189/18722 1.158448e-116 1.630516e-113 1.093209e-113
## GO:0046034 277/18722 3.636826e-111 3.412555e-108 2.288010e-108
## GO:0045333 230/18722 1.228519e-110 8.645704e-108 5.796671e-108
## GO:0019646  87/18722 7.658099e-109 4.311510e-106 2.890731e-106
## GO:0042773  95/18722 1.128535e-107 4.538322e-105 3.042801e-105
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              geneID
## GO:0006119                                             ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRH/UQCRHL/UQCRQ
## GO:0009060                                    ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/IDE/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHB/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRH/UQCRHL/UQCRQ
## GO:0046034 APP/ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5MC1/ATP5MC2/ATP5MC3/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BAD/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/GAPDH/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/PSEN1/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRH/UQCRHL/UQCRQ
## GO:0045333                            ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/IDE/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHB/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0019646                                                                                                                                                       COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6C/COX7A1/COX7A2/COX7A2L/COX7C/COX8A/COX8C/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRH/UQCRHL/UQCRQ
## GO:0042773                                                                                                                                                   BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6C/COX7A1/COX7A2/COX7A2L/COX7C/COX8A/COX8C/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRH/UQCRHL/UQCRQ
##            Count
## GO:0006119    75
## GO:0009060    77
## GO:0046034    82
## GO:0045333    78
## GO:0019646    60
## GO:0042773    61
</pre></div>
</div>
<p>Then, we can quickly get a glimpse of the main results.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>result %&gt;% dotplot()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-4-11.png" /></p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>result %&gt;% cnetplot()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-5-11.png" /></p>
<p>Now, we would like to see the overlap between the genes in the top 5
enriched categories.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ComplexHeatmap)

# extract lists of genes in top enriched terms
terms_oi = result %&gt;% as.data.frame() %&gt;% filter(p.adjust&lt;0.05) %&gt;% slice_max(Count, n=10)
terms_oi = terms_oi %&gt;% group_by(Description) %&gt;% summarize(genes=strsplit(geneID, split=&quot;/&quot;)) %&gt;% ungroup()
terms_oi = sapply(terms_oi[[&quot;Description&quot;]], 
                  function(genes){ terms_oi[[&quot;genes&quot;]][[1]]}, simplify=FALSE)

# prepare inputs
terms_oi = terms_oi %&gt;% 
    list_to_matrix() %&gt;% 
    make_comb_mat()
  
# plot
plt = terms_oi %&gt;% UpSet()
plt
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-6-11.png" /></p>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="phylogenetic-trees-with-ggtree">
<h1>Phylogenetic trees with <code class="docutils literal notranslate"><span class="pre">ggtree</span></code><a class="headerlink" href="#phylogenetic-trees-with-ggtree" title="Permalink to this headline">¶</a></h1>
<p>Trees are another very common plot to visualize hierarchical patterns.
Here, we will use
<a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a>
to visualize the phylogenetic relationships extracted from a multiple
sequence alignment of <em>TP53</em> across mammals.</p>
<div class="section" id="basic-tree">
<h2>Basic tree<a class="headerlink" href="#basic-tree" title="Permalink to this headline">¶</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ggtree)
set.seed(100)
tree &lt;- rtree(50)
ggtree(tree, layout=&quot;circular&quot;) + geom_tiplab()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-7-11.png" /></p>
</div>
<div class="section" id="tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
<h2>Tree and multiple sequence alignment of <em>TP53</em> amino acids<a class="headerlink" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids" title="Permalink to this headline">¶</a></h2>
<p>We will follow <a class="reference external" href="https://github.com/RussellGrayxd/Phylogenetics/blob/master/Sequence%20analysis.R">Russell J.
Gray</a>’s
approach.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(seqinr)
require(ape)

# create tree from alignment
fasta_file = file.path(here::here(), &quot;data&quot;, &quot;raw&quot;, &quot;TP53-mammals-alignment-aa.fa&quot;)
aln = read.alignment(fasta_file, format = &quot;fasta&quot;, whole.header=TRUE)
D = dist.alignment(aln)
tree = njs(D)

# plot tree with MSA
tree_plot = ggtree(tree)
msaplot(tree_plot, fasta = fasta_file) + ggtitle(&quot;tree with MSA&quot;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-8-11.png" /></p>
</div>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="references">
<h1>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_ALZHEIMERS_DISEASE">Gene set
used</a></p></li>
<li><p><a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a></p></li>
<li><p><a class="reference external" href="https://seqinr.r-forge.r-project.org/"><code class="docutils literal notranslate"><span class="pre">seqinr</span></code></a></p></li>
<li><p><a class="reference external" href="https://www.rdocumentation.org/packages/ape/versions/5.4-1"><code class="docutils literal notranslate"><span class="pre">ape</span></code></a></p></li>
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html"><code class="docutils literal notranslate"><span class="pre">Biostrings</span></code></a></p></li>
<li><p><a class="reference external" href="https://github.com/RussellGrayxd"><code class="docutils literal notranslate"><span class="pre">Russell</span> <span class="pre">J.</span> <span class="pre">Gray's</span> <span class="pre">github</span></code></a></p></li>
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html"><code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code></a></p></li>
</ul>
</div>
<div class="tex2jax_ignore mathjax_ignore section" id="session-info">
<h1>Session Info<a class="headerlink" href="#session-info" title="Permalink to this headline">¶</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>sessionInfo()

## R version 4.1.2 (2021-11-01)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ape_5.6-2             seqinr_4.2-8          ggtree_3.2.1         
##  [4] ComplexHeatmap_2.10.0 org.Hs.eg.db_3.14.0   AnnotationDbi_1.56.2 
##  [7] IRanges_2.28.0        S4Vectors_0.32.3      Biobase_2.54.0       
## [10] BiocGenerics_0.40.0   clusterProfiler_4.2.2 forcats_0.5.1        
## [13] stringr_1.4.0         dplyr_1.0.8           purrr_0.3.4          
## [16] readr_2.1.2           tidyr_1.2.0           tibble_3.1.6         
## [19] ggplot2_3.3.5         tidyverse_1.3.1      
## 
## loaded via a namespace (and not attached):
##   [1] circlize_0.4.14        readxl_1.3.1           shadowtext_0.1.1      
##   [4] backports_1.4.1        fastmatch_1.1-3        plyr_1.8.6            
##   [7] igraph_1.2.11          lazyeval_0.2.2         splines_4.1.2         
##  [10] BiocParallel_1.28.3    GenomeInfoDb_1.30.1    digest_0.6.29         
##  [13] foreach_1.5.2          yulab.utils_0.0.4      htmltools_0.5.2       
##  [16] GOSemSim_2.20.0        viridis_0.6.2          GO.db_3.14.0          
##  [19] fansi_1.0.2            magrittr_2.0.2         memoise_2.0.1         
##  [22] cluster_2.1.2          doParallel_1.0.17      tzdb_0.2.0            
##  [25] Biostrings_2.62.0      graphlayouts_0.8.0     modelr_0.1.8          
##  [28] matrixStats_0.61.0     enrichplot_1.14.2      colorspace_2.0-3      
##  [31] blob_1.2.2             rvest_1.0.2            ggrepel_0.9.1         
##  [34] haven_2.4.3            xfun_0.30              crayon_1.5.0          
##  [37] RCurl_1.98-1.6         jsonlite_1.8.0         scatterpie_0.1.7      
##  [40] iterators_1.0.14       glue_1.6.2             polyclip_1.10-0       
##  [43] gtable_0.3.0           zlibbioc_1.40.0        XVector_0.34.0        
##  [46] GetoptLong_1.0.5       shape_1.4.6            scales_1.1.1          
##  [49] DOSE_3.20.1            DBI_1.1.2              Rcpp_1.0.8            
##  [52] viridisLite_0.4.0      clue_0.3-60            gridGraphics_0.5-1    
##  [55] tidytree_0.3.9         bit_4.0.4              httr_1.4.2            
##  [58] fgsea_1.20.0           RColorBrewer_1.1-2     ellipsis_0.3.2        
##  [61] pkgconfig_2.0.3        farver_2.1.0           dbplyr_2.1.1          
##  [64] utf8_1.2.2             here_1.0.1             ggplotify_0.1.0       
##  [67] tidyselect_1.1.2       labeling_0.4.2         rlang_1.0.2           
##  [70] reshape2_1.4.4         munsell_0.5.0          cellranger_1.1.0      
##  [73] tools_4.1.2            cachem_1.0.6           downloader_0.4        
##  [76] cli_3.2.0              generics_0.1.2         RSQLite_2.2.10        
##  [79] ade4_1.7-18            broom_0.7.12           evaluate_0.15         
##  [82] fastmap_1.1.0          yaml_2.3.5             knitr_1.37            
##  [85] bit64_4.0.5            fs_1.5.2               tidygraph_1.2.0       
##  [88] KEGGREST_1.34.0        ggraph_2.0.5           nlme_3.1-155          
##  [91] aplot_0.1.2            DO.db_2.9              xml2_1.3.3            
##  [94] compiler_4.1.2         rstudioapi_0.13        png_0.1-7             
##  [97] reprex_2.0.1           treeio_1.18.1          tweenr_1.0.2          
## [100] stringi_1.7.6          highr_0.9              lattice_0.20-45       
## [103] Matrix_1.4-0           vctrs_0.3.8            pillar_1.7.0          
## [106] lifecycle_1.0.1        GlobalOptions_0.1.2    data.table_1.14.2     
## [109] bitops_1.0-7           patchwork_1.1.1        qvalue_2.26.0         
## [112] R6_2.5.1               gridExtra_2.3          codetools_0.2-18      
## [115] MASS_7.3-55            assertthat_0.2.1       rjson_0.2.21          
## [118] rprojroot_2.0.2        withr_2.5.0            GenomeInfoDbData_1.2.7
## [121] parallel_4.1.2         hms_1.1.1              ggfun_0.0.5           
## [124] rmarkdown_2.12         ggforce_0.3.3          lubridate_1.8.0
</pre></div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="02-tidytuesday.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Customize every detail</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="04-exercise.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Exploratory Data Analysis of the most popular human gene</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Miquel Anglada Girotto (miquel [dot] anglada [at] crg [dot] eu)<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>