

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Biology-related data visualization &#8212; Practical tools for quick data visualization</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '04-bio';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="README.html">
  
  
  
  
  
  
    <p class="title logo__title">Practical tools for quick data visualization</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="README.html">
                    Practical tools for quick data visualization
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01-intro.html">Get started</a></li>





<li class="toctree-l1"><a class="reference internal" href="02-tidytuesday.html">Customize every detail</a></li>





<li class="toctree-l1"><a class="reference internal" href="03-bio.html">Biology-related data visualization</a></li>




<li class="toctree-l1"><a class="reference internal" href="04-exercise.html">Exploratory Data Analysis of the most popular human gene</a></li>




</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/MiqG/practical_tools_for_quick_data_visualization" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/MiqG/practical_tools_for_quick_data_visualization/issues/new?title=Issue%20on%20page%20%2F04-bio.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/04-bio.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Biology-related data visualization</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Biology-related data visualization</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler">Gene Set Enrichment Analysis (GSEA) with <code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code></a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">Phylogenetic trees with <code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#basic-tree">Basic tree</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">Tree and multiple sequence alignment of <em>TP53</em> amino acids</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#session-info">Session Info</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <div class="section" id="biology-related-data-visualization">
<h1>Biology-related data visualization<a class="headerlink" href="#biology-related-data-visualization" title="Permalink to this heading">#</a></h1>
</div>
<div class="section" id="gene-set-enrichment-analysis-gsea-with-clusterprofiler">
<h1>Gene Set Enrichment Analysis (GSEA) with <code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code><a class="headerlink" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler" title="Permalink to this heading">#</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(tidyverse)
</pre></div>
</div>
<p>Let’s say we have found a set of interesting genes.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>geneset_file = file.path(here::here(), &quot;data&quot;, &quot;raw&quot;, &quot;geneset.txt&quot;)
genes_oi = readLines(geneset_file)

genes_oi %&gt;% length() %&gt;% print()

## [1] 166

genes_oi %&gt;% head()

## [1] &quot;ADAM10&quot; &quot;ADAM17&quot; &quot;APAF1&quot;  &quot;APBB1&quot;  &quot;APH1A&quot;  &quot;APOE&quot;
</pre></div>
</div>
<p>Is is enriched in some biological processes?</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(clusterProfiler)
require(org.Hs.eg.db)

result = enrichGO(genes_oi, ont=&quot;BP&quot;, keyType=&quot;SYMBOL&quot;, OrgDb=org.Hs.eg.db)
result %&gt;% as.data.frame() %&gt;% head()

##                    ID                                            Description
## GO:0006119 GO:0006119                              oxidative phosphorylation
## GO:0042775 GO:0042775 mitochondrial ATP synthesis coupled electron transport
## GO:0042773 GO:0042773               ATP synthesis coupled electron transport
## GO:0046034 GO:0046034                                  ATP metabolic process
## GO:0022904 GO:0022904                   respiratory electron transport chain
## GO:0022900 GO:0022900                               electron transport chain
##            GeneRatio   BgRatio        pvalue      p.adjust        qvalue
## GO:0006119    79/157 148/18862 2.595331e-132 7.342192e-129 4.909274e-129
## GO:0042775    62/157  99/18862 1.842964e-108 2.606873e-105 1.743056e-105
## GO:0042773    62/157 100/18862 4.825741e-108 3.501476e-105 2.341222e-105
## GO:0046034    83/157 313/18862 4.950831e-108 3.501476e-105 2.341222e-105
## GO:0022904    63/157 117/18862 3.453413e-104 1.953941e-101 1.306481e-101
## GO:0022900    68/157 178/18862  1.378807e-99  6.501077e-97  4.346872e-97
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      geneID
## GO:0006119                     ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5MC1/ATP5MC2/ATP5MC3/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0042775                                                                                                                                                   BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6C/COX7A2L/COX7B/COX7C/COX8A/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0042773                                                                                                                                                   BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6C/COX7A2L/COX7B/COX7C/COX8A/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0046034 APP/ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5MC1/ATP5MC2/ATP5MC3/ATP5PB/ATP5PD/ATP5PF/ATP5PO/BAD/BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6B2/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/GAPDH/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/PSEN1/SDHA/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0022904                                                                                                                                              BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6C/COX7A2L/COX7B/COX7C/COX8A/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHB/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
## GO:0022900                                                                                                          BID/COX4I1/COX4I2/COX5A/COX5B/COX6A1/COX6A2/COX6B1/COX6C/COX7A1/COX7A2/COX7A2L/COX7B/COX7B2/COX7C/COX8A/COX8C/CYC1/CYCS/NDUFA1/NDUFA10/NDUFA2/NDUFA3/NDUFA4/NDUFA4L2/NDUFA5/NDUFA6/NDUFA7/NDUFA8/NDUFA9/NDUFAB1/NDUFB1/NDUFB10/NDUFB2/NDUFB3/NDUFB4/NDUFB5/NDUFB6/NDUFB7/NDUFB8/NDUFB9/NDUFC1/NDUFC2/NDUFS1/NDUFS2/NDUFS3/NDUFS4/NDUFS5/NDUFS6/NDUFS7/NDUFS8/NDUFV1/NDUFV2/NDUFV3/SDHA/SDHB/SDHC/SDHD/SNCA/UQCR10/UQCR11/UQCRB/UQCRC1/UQCRC2/UQCRFS1/UQCRH/UQCRHL/UQCRQ
##            Count
## GO:0006119    79
## GO:0042775    62
## GO:0042773    62
## GO:0046034    83
## GO:0022904    63
## GO:0022900    68
</pre></div>
</div>
<p>Then, we can quickly get a glimpse of the main results.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>result %&gt;% dotplot()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-4-13.png" /></p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>result %&gt;% cnetplot()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-5-13.png" /></p>
<p>Now, we would like to see the overlap between the genes in the top 5
enriched categories.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ComplexHeatmap)

# extract lists of genes in top enriched terms
terms_oi = result %&gt;% as.data.frame() %&gt;% filter(p.adjust&lt;0.05) %&gt;% slice_max(Count, n=10)
terms_oi = 
  terms_oi %&gt;% 
  group_by(Description) %&gt;% 
  summarize(genes=strsplit(geneID, split=&quot;/&quot;)) %&gt;% 
  ungroup() %&gt;%
  deframe()

# prepare inputs
terms_oi = terms_oi %&gt;% 
    list_to_matrix() %&gt;% 
    make_comb_mat()
  
# plot
plt = terms_oi %&gt;% UpSet()
plt
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-6-13.png" /></p>
</div>
<div class="section" id="phylogenetic-trees-with-ggtree">
<h1>Phylogenetic trees with <code class="docutils literal notranslate"><span class="pre">ggtree</span></code><a class="headerlink" href="#phylogenetic-trees-with-ggtree" title="Permalink to this heading">#</a></h1>
<p>Trees are another very common plot to visualize hierarchical patterns.
Here, we will use
<a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a>
to visualize the phylogenetic relationships extracted from a multiple
sequence alignment of <em>TP53</em> across mammals.</p>
<div class="section" id="basic-tree">
<h2>Basic tree<a class="headerlink" href="#basic-tree" title="Permalink to this heading">#</a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(ggtree)
set.seed(100)
tree &lt;- rtree(50)
ggtree(tree, layout=&quot;circular&quot;) + geom_tiplab()
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-7-13.png" /></p>
</div>
<div class="section" id="tree-and-multiple-sequence-alignment-of-tp53-amino-acids">
<h2>Tree and multiple sequence alignment of <em>TP53</em> amino acids<a class="headerlink" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids" title="Permalink to this heading">#</a></h2>
<p>We will follow <a class="reference external" href="https://github.com/RussellGrayxd/Phylogenetics/blob/master/Sequence%20analysis.R">Russell J.
Gray</a>’s
approach.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>require(seqinr)
require(ape)

# create tree from alignment
fasta_file = file.path(here::here(), &quot;data&quot;, &quot;raw&quot;, &quot;TP53-mammals-alignment-aa.fa&quot;)
aln = read.alignment(fasta_file, format = &quot;fasta&quot;, whole.header=TRUE)
D = dist.alignment(aln)
tree = njs(D)

# plot tree with MSA
tree_plot = ggtree(tree)
msaplot(tree_plot, fasta = fasta_file) + ggtitle(&quot;tree with MSA&quot;)
</pre></div>
</div>
<p><img alt="" src="_images/unnamed-chunk-8-13.png" /></p>
</div>
</div>
<div class="section" id="references">
<h1>References<a class="headerlink" href="#references" title="Permalink to this heading">#</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_ALZHEIMERS_DISEASE">Gene set
used</a></p></li>
<li><p><a class="reference external" href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html"><code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a></p></li>
<li><p><a class="reference external" href="https://seqinr.r-forge.r-project.org/"><code class="docutils literal notranslate"><span class="pre">seqinr</span></code></a></p></li>
<li><p><a class="reference external" href="https://www.rdocumentation.org/packages/ape/versions/5.4-1"><code class="docutils literal notranslate"><span class="pre">ape</span></code></a></p></li>
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html"><code class="docutils literal notranslate"><span class="pre">Biostrings</span></code></a></p></li>
<li><p><a class="reference external" href="https://github.com/RussellGrayxd"><code class="docutils literal notranslate"><span class="pre">Russell</span> <span class="pre">J.</span> <span class="pre">Gray's</span> <span class="pre">github</span></code></a></p></li>
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html"><code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code></a></p></li>
</ul>
</div>
<div class="section" id="session-info">
<h1>Session Info<a class="headerlink" href="#session-info" title="Permalink to this heading">#</a></h1>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>sessionInfo()

## R version 4.3.1 (2023-06-16)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so;  LAPACK version 3.7.1
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Europe/Madrid
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ape_5.7-1             seqinr_4.2-36         ggtree_3.0.4         
##  [4] ComplexHeatmap_2.9.3  org.Hs.eg.db_3.13.0   AnnotationDbi_1.54.1 
##  [7] IRanges_2.30.1        S4Vectors_0.40.1      Biobase_2.52.0       
## [10] BiocGenerics_0.48.1   clusterProfiler_4.0.5 lubridate_1.9.2      
## [13] forcats_1.0.0         stringr_1.5.1         dplyr_1.1.1          
## [16] purrr_0.3.4           readr_2.1.5           tidyr_1.2.1          
## [19] tibble_3.2.1          ggplot2_3.5.0         tidyverse_2.0.0      
## 
## loaded via a namespace (and not attached):
##   [1] RColorBrewer_1.1-3     shape_1.4.6.1          jsonlite_1.8.0        
##   [4] magrittr_2.0.3         magick_2.8.3           farver_2.1.1          
##   [7] rmarkdown_2.26         GlobalOptions_0.1.2    fs_1.6.2              
##  [10] zlibbioc_1.38.0        vctrs_0.6.1            memoise_2.0.1         
##  [13] Cairo_1.6-1            RCurl_1.98-1.14        htmltools_0.5.7       
##  [16] gridGraphics_0.5-1     plyr_1.8.9             cachem_1.0.8          
##  [19] igraph_1.3.1           lifecycle_1.0.4        iterators_1.0.14      
##  [22] pkgconfig_2.0.3        Matrix_1.6-5           R6_2.5.1              
##  [25] fastmap_1.1.1          clue_0.3-65            GenomeInfoDbData_1.2.6
##  [28] digest_0.6.29          aplot_0.2.2            enrichplot_1.12.2     
##  [31] colorspace_2.0-3       patchwork_1.2.0        rprojroot_2.0.4       
##  [34] RSQLite_2.3.5          labeling_0.4.3         fansi_1.0.3           
##  [37] timechange_0.3.0       httr_1.4.7             polyclip_1.10-6       
##  [40] compiler_4.3.1         here_1.0.1             bit64_4.0.5           
##  [43] withr_3.0.0            doParallel_1.0.17      downloader_0.4        
##  [46] BiocParallel_1.26.2    viridis_0.6.5          DBI_1.2.2             
##  [49] highr_0.10             ggforce_0.4.2          MASS_7.3-60           
##  [52] rjson_0.2.21           tools_4.3.1            DO.db_2.9             
##  [55] scatterpie_0.2.1       glue_1.6.2             nlme_3.1-164          
##  [58] GOSemSim_2.18.1        shadowtext_0.1.3       ade4_1.7-22           
##  [61] cluster_2.1.4          reshape2_1.4.4         fgsea_1.18.0          
##  [64] generics_0.1.3         gtable_0.3.4           tzdb_0.3.0            
##  [67] data.table_1.15.2      hms_1.1.3              tidygraph_1.2.0       
##  [70] utf8_1.2.4             XVector_0.32.0         ggrepel_0.9.5         
##  [73] foreach_1.5.2          pillar_1.9.0           yulab.utils_0.1.4     
##  [76] circlize_0.4.16        splines_4.3.1          tweenr_2.0.3          
##  [79] treeio_1.16.2          lattice_0.21-8         bit_4.0.5             
##  [82] tidyselect_1.2.1       GO.db_3.13.0           Biostrings_2.64.1     
##  [85] knitr_1.45             gridExtra_2.3          xfun_0.42             
##  [88] graphlayouts_1.1.1     matrixStats_1.0.0      stringi_1.7.8         
##  [91] lazyeval_0.2.2         ggfun_0.1.4            yaml_2.3.8            
##  [94] evaluate_0.23          codetools_0.2-19       ggraph_2.2.1          
##  [97] qvalue_2.24.0          ggplotify_0.1.2        cli_3.4.0             
## [100] munsell_0.5.0          Rcpp_1.0.8.3           GenomeInfoDb_1.28.4   
## [103] png_0.1-8              parallel_4.3.1         blob_1.2.4            
## [106] DOSE_3.18.2            bitops_1.0-7           viridisLite_0.4.2     
## [109] tidytree_0.4.6         scales_1.3.0           crayon_1.5.2          
## [112] GetoptLong_1.0.5       rlang_1.1.0            cowplot_1.1.3         
## [115] fastmatch_1.1-4        KEGGREST_1.32.0
</pre></div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Biology-related data visualization</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-set-enrichment-analysis-gsea-with-clusterprofiler">Gene Set Enrichment Analysis (GSEA) with <code class="docutils literal notranslate"><span class="pre">clusterProfiler</span></code></a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#phylogenetic-trees-with-ggtree">Phylogenetic trees with <code class="docutils literal notranslate"><span class="pre">ggtree</span></code></a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#basic-tree">Basic tree</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#tree-and-multiple-sequence-alignment-of-tp53-amino-acids">Tree and multiple sequence alignment of <em>TP53</em> amino acids</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#session-info">Session Info</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Miquel Anglada Girotto (miquel [dot] anglada [at] crg [dot] eu)
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>